doc/source/text.rst

```diff
@@ -205,6 +217,52 @@ and optional groups like
 
 can also be used.
 
+When ``str.extract`` method is used on an ``Index``, it returns a
+``DataFrame`` with as many columns as capture groups.
```

Comment:
and if the capture group has a ``?P<...>`` name then it will be named like this, otherwise, numbered for its position?